frontend/src/pages/NewRun.test.tsx

```diff
@@ -0,0 +1,1160 @@
+
+      expect(tree.find('#createNewRunBtn').props()).toHaveProperty('disabled', true);
+      tree.unmount();
+    });
+
+    it('enables the \'Create\' new run button if pipeline ID in query params and run name entered', async () => {
+      const props = generateProps();
+      props.location.search = `?${QUERY_PARAMS.pipelineId}=${newMockPipeline().id}`;
+
+      const tree = shallow(<TestNewRun {...props as any} />);
+      (tree.instance() as any).handleChange('runName')({ target: { value: 'run name' } });
+      await TestUtils.flushPromises();
+
+      expect(tree.find('#createNewRunBtn').props()).toHaveProperty('disabled', false);
+      tree.unmount();
+    });
+
+    it('re-disables the \'Create\' new run button if pipeline ID in query params and run name entered then cleared', async () => {
+      const props = generateProps();
+      props.location.search = `?${QUERY_PARAMS.pipelineId}=${newMockPipeline().id}`;
+
+      const tree = shallow(<TestNewRun {...props as any} />);
+      (tree.instance() as any).handleChange('runName')({ target: { value: 'run name' } });
+      await TestUtils.flushPromises();
+      expect(tree.find('#createNewRunBtn').props()).toHaveProperty('disabled', false);
+
+      (tree.instance() as any).handleChange('runName')({ target: { value: '' } });
+      expect(tree.find('#createNewRunBtn').props()).toHaveProperty('disabled', true);
+
+      tree.unmount();
+    });
+
+    it('sends a request to create a new run when \'Create\' is clicked', async () => {
+      const props = generateProps();
+      const experiment = newMockExperiment();
+      const pipeline = newMockPipeline();
+      props.location.search =
+        `?${QUERY_PARAMS.experimentId}=${experiment.id}`
+        + `&${QUERY_PARAMS.pipelineId}=${pipeline.id}`;
+
+      getExperimentSpy.mockImplementation(() => experiment);
+      getPipelineSpy.mockImplementation(() => pipeline);
+
+      const tree = shallow(<TestNewRun {...props as any} />);
+      (tree.instance() as any).handleChange('runName')({ target: { value: 'test run name' } });
+      (tree.instance() as any).handleChange('description')({ target: { value: 'test run description' } });
+      await TestUtils.flushPromises();
+
+      tree.find('#createNewRunBtn').simulate('click');
+      // The create APIs are called in a callback triggered by clicking 'Create', so we wait again
+      await TestUtils.flushPromises();
+
+      expect(createRunSpy).toHaveBeenCalledTimes(1);
+      expect(createRunSpy).toHaveBeenLastCalledWith({
+        description: 'test run description',
+        name: 'test run name',
+        pipeline_spec: {
+          parameters: pipeline.parameters,
+          pipeline_id: pipeline.id,
+        },
+        resource_references: [{
+          key: {
+            id: experiment.id,
+            type: ApiResourceType.EXPERIMENT,
+          },
+          relationship: ApiRelationship.OWNER,
+        }]
+      });
+      tree.unmount();
+    });
+
```

Comment:
Add test to make sure selecting a new pipeline updates the parameter fields? Also that selecting a pipeline with no parameters removes all parameter fields?